### ISSUE TITLE
Inexhaustible granite mines broken

### ISSUE THREAD

**Issue by Spikeone** [Return-To-The-Roots/s25client#918]
Sadly that addon is broken. It is intended to allow the player to build a granite mine anywhere - regardless of actual resources. In v20160228 it was still working as expected.

**Comment by Xermio** [comment]
The addon description is "Granite mines will never be depleted.".

This does not mean that you can build a (working) granite mine everywhere?

I understand it as: The mine will only work on locations with this resource, but they will never deplete.

The description should be updated if this addon is intended to allow granite mines everywhere.

**Comment by Spikeone** [comment]
Should be updated as well, thats right. Intention of the addon is: everywhere you mine, you will always find stones for building. 

Also this is needed for quite long games where you have to rebuild stuff often.

**Comment by Flamefire** [comment]
@Xermio is right: The addon is a special case of the inexhaustible mines addon that only applies to granite mines: It simply makes the granite mines not "consume" the resource in the mountain.

IMO allowing to build granite mines just anywhere is so "wrong" (against the original game and any logic) that I would not do this. Of course as it is an addon I can change it if the demand is there.

**Comment by Xermio** [comment]
I think it would be very helpful for the AI. The AI has very often no more stones.

**Comment by Flamefire** [comment]
It would need to be adapted to know this. I'd rather would make the AI build granite mines more often and make them inexhaustible.

With this suggestion the AI would need to build mines where no resource is...

**Comment by Spikeone** [comment]
Actually as this is the one and only Addon I made I should know what the intention was behind ;)

And it was: build a granite mine anywhere you want so when playing large maps you always have stone even if the mapper did not include granite at all. It was not an addition to inexhaustible mines but merely a mix of "granite everywhere" and "granite mine only inexhaustible".

So if anyone wants a "granite mine inexhaustible only" addon thats fine but its not this one.

As said working correctly in v20160228

**Comment by Flamefire** [comment]
Ouch :D

Well the name and description was very misleading then, I'll double-check the code where it broke. Would you mind coming up with a better name and description?

**Comment by Spikeone** [comment]
Actually I can't remember giving it a name nor a description, but what about:

Title: Granite mines always mine
or
Title: Granite mines need no resource
or
Title: Mountains always have granite

Desc: Granite mines will need no granite resource for working and will never deplete
or
Desc: Mountains always contain granite which can be mined with granite mines wich never depletes
or
Desc: Any mine spot has granite, regardless of the current resource. Furthermore granite mines will not deplete anymore.

This addon might be improved as geologists could find "granite" when no other resource is found anymore and this addon is enabled.

**Comment by Flamefire** [comment]
What about the logic issue? You can harvest something that is not there. Especially for AI this is hard. I do think this strays to far from S2 as with the same ideas you could make woodcutters that don't need trees etc.

With your suggestion:
>This addon might be improved as geologists could find "granite" when no other resource is found anymore and this addon is enabled.

Idea: The addon simply adds full granite to all empty spots (during the initial map load, just like the no gold addon does) *and* granite mines are inexhaustible.

**Comment by Spikeone** [comment]
Well, that doesn't help on most maps where you nearly have no mountain spot without resource. One also has to add granite at positions where any resource has been used while playing - then it'd work. 

And no, I can't agree that there is a major logic flaw (although it might depend on how you look at it): If there is no resource in a mountain, there still is a mountain - which is made of stone. When you have no tree, then there is just sand.

Same goes for other addons... ships magically are faster? Trees spawn animals? I burn down a mine instead of just leaving it? A 99% finished building is just destroyed but 100% makes it burn? Most things of settlers 2 has logic flaws, but for the good of gameplay so thats no argument I will follow. As this addon is thought for maps which take long times I don't see any reason to watch out for some logic that doesn't even change something at all.

As this is said: If it helps anyhow just replace nothing with granite, when a spot is emptied just fill it with granite and make granite mines use no resources. Still has the same result for what I need this addon for and everybody is hopefully happy :)

**Comment by Flamefire** [comment]
>  If there is no resource in a mountain, there still is a mountain - which is made of stone. When you have no tree, then there is just sand.

Sure. I just meant the flaw that there IS a mechanism for mining stone at points where this resource exists on a mountain. This mechanism is part of S2: You can only get resources (wood, stone, coal, ...) when there is some at that spot. This addon substantially changes this.

The other addons just slightly change things: Faster ships (maybe better sails?), trees always spawn animals (actually it is that in the forest animals live and reproduce, you just change the fertility), 99% vs 100% burn (well that is S2 logic, we didn't change it) etc.

You are right about AI, it assumes the intended behaviour. As replacing empty spots DURING the game will also have difficulties (e.g. endless mines addon enabled) I'd rather either change them during map init or not at all.

If you really see no problem with the changed logic I'll do it that way, although I don't like it... But well, people will have different opinions and no ones is forced to use this.

**Comment by Spikeone** [comment]
> The other addons just slightly change things: Faster ships (maybe better sails?), trees always spawn animals (actually it is that in the forest animals live and reproduce, you just change the fertility), 99% vs 100% burn (well that is S2 logic, we didn't change it) etc.

Yes, I didn't mean addons but just settlers 2, that it has some flaws like wells create buckets out of nowhere, soldiers wear armor but you produce none etc. So it's more about the general gameplay / feeling. At least for me settlers is a lot about the feeling: Produce some simple stuff, put it in a building and get something more valuable. First you can get food by building hunter / fisher but later you need bigger production buildings. As we can not create resources (as this breaks some maps) we have a general game design flaw. It was never meant for longer games that are balanced I think. For that reason we created the charburner which is more expensive than coal mines. 

We could do the same (later) for granite, instead of my granite mine addon one could add a building that is just placed over the ground and mines down (e.g. normal building). That way we don't have the "logic flaw" of mining resources one does not have and could use other resources for production (boards + food, boards for reinforcing tunnels?). I actually like this idea as well.

But yes, I see no problem with the changed logic as for me it always was a flaw that a mountain needs stone to... have stone. But as you already said, if one doesn't want that flaw, one doesn't use the addon. Still adding those other ideas (replace nothing with stone like we do with gold) and an additional building would be nice, wouldn't it?

So in conclusion: yes please change this addon to the old behavior :)
